### DISCUSSION TITLE
AI Smart Assistant

### DISCUSSION THREAD

**Discussion by hAzUpL8** [dbeaver/dbeaver#19142] [Q&A] [answered]
### Description

It is more like a question. 
Does DBeaver allways sends the database metadata to the OpenAI platform along side with my query?
I got this error on every of my prompta even if they are very small Like "List me all duplicate records of table?"

"This model's maximum context length is 8001 tokens, however you requested 9574 tokens (8574 in your prompt; 1000 for the completion). Please reduce your prompt; or completion length."

My prompt is not 8574 tokens long!

### DBeaver Version

Community edition Version 22.3.4.202302060909

### Operating System

Windows 10

### Database and driver

Oracle JDBC Thin Client

### Steps to reproduce

Write a prompt "List me all duplicate records of table?" in GPT-3 smart completition window

**Comment by E1izabeth** [comment]
Duplicate of #19063
Yes DBeaver send metadata with each query. I believe you can find answer about how to solve your issue here https://github.com/dbeaver/dbeaver/issues/19063#issuecomment-1433164027.

**Comment by Destrolaric** [comment, marked as answer]
@hAzUpL8 Unfortunately, this is the case, GPT can't memorize metadata between requests, so the metadata is sent every request. We plan to improve this by allowing a more detailed selection of entity metadata to send.
